### DISCUSSION TITLE
Micronaut Framework 4.5.0 Plan

### DISCUSSION THREAD

**Discussion by sdelamo** [micronaut-projects/micronaut-core#10786] [Planning]
**Planned Date: 2024-05-29**

This document describes the next minor release of the Micronaut Framework. To learn more about the release process, visit the  [Micronaut Roadmap](https://micronaut.io/micronaut-roadmap/) page. 

## Advocacy
We already planned this in the [4.4.0 plan](https://github.com/micronaut-projects/micronaut-core/discussions/10527) but failed to execute it fully. The plan is to put a significant emphasis on the advocacy side in the 4.5.0 release. 

## Youtube Channel
Release ten videos of less than 15m for the YouTube channel covering features of the framework. We have to improve our presence on YouTube. Many people only learn with video nowadays.

## Guides
Release 4 Guides comparing Micronaut Framework to Spring Boot. It is vital to show how easy it is to move from Spring Boot to Micronaut Framework.

## New Modules
I think we may be able to release several modules: 

### [SourceGen](https://github.com/micronaut-projects/micronaut-sourcegen/issues)
Micronaut SourceGen provides a fork for JavaPoet with enhanced features and abstraction for performing source code generation across languages.

### [JSON Schema](https://github.com/micronaut-projects/micronaut-json-schema/)
Micronaut JSON Schema allows creating schemas at compilation-time for beans in your applications.

Given: 

![java-json-schema](https://github.com/micronaut-projects/micronaut-core/assets/864788/cfbfaf5d-880e-4a73-bd5b-726321fc10d3)

The application will generate at compile-time: 
![json](https://github.com/micronaut-projects/micronaut-core/assets/864788/dab8ed72-94af-4751-a4ac-c48cd96cd786)

### Cursored Pagination for SQL
We have already a [PR #2884](https://github.com/micronaut-projects/micronaut-data/pull/2884) in progress.
### Multitenancy-Discriminator Column
We have already a [PR #2876](https://github.com/micronaut-projects/micronaut-data/pull/2876) almost ready.  

The idea is you can annotate a field in an Entity as the discriminator column: 
![multitenancy](https://github.com/micronaut-projects/micronaut-core/assets/864788/3ada6276-4a2f-4f02-80e5-d72e620e9f77)

## Backlog
If you think we should prioritize something else, don’t hesitate to leave a comment below.

The [RoadMap top-level project](https://github.com/orgs/micronaut-projects/projects/69) lists features and ideas that may happen in future releases.

**Comment by nbrugger-tgm** [comment]
Hi i have 3 very closely related bugs that are quite fundamental to `@Factory`, AOP and introspection: #10836,  #10834 and #10833

Since this is a issue of `core` and (especially  #10836) cause "invisible" side effects (validations dont apply to `@Factory` generated bean classes). So its possible that more people are affected without knowing it, since it is "only" a "you do not get Validation exceptions"
